src/shared/bus-wait-for-jobs.c

```diff
@@ -161,28 +161,31 @@ static int bus_job_get_service_result(BusWaitForJobs *d, char **ret) {
                                           ret);
 }
 
-static void log_job_error_with_service_result(
+static int log_job_error_with_service_result(
                 const char* service,
                 const char *result,
                 bool quiet,
                 const char* const* extra_args) {
 
         static const struct {
-                const char *result, *explanation;
+                const char *result;
+                int error;
+                const char *explanation;
         } explanations[] = {
-                { "resources",       "of unavailable resources or another system error"                      },
-                { "protocol",        "the service did not take the steps required by its unit configuration" },
-                { "timeout",         "a timeout was exceeded"                                                },
-                { "exit-code",       "the control process exited with error code"                            },
-                { "signal",          "a fatal signal was delivered to the control process"                   },
-                { "core-dump",       "a fatal signal was delivered causing the control process to dump core" },
-                { "watchdog",        "the service failed to send watchdog ping"                              },
-                { "start-limit-hit", "start of the service was attempted too often"                          },
-                { "oom-kill",        "of an out-of-memory (OOM) siutation"                                   },
+                { "resources",       EBUSY,           "of unavailable resources or another system error"                      },
+                { "protocol",        EPROTONOSUPPORT, "the service did not take the steps required by its unit configuration" },
+                { "timeout",         ETIMEDOUT,       "a timeout was exceeded"                                                },
+                { "exit-code",       EPIPE,           "the control process exited with error code"                            },
+                { "signal",          EINTR,           "a fatal signal was delivered to the control process"                   },
```

Comment:
i would avoid EINTR, because it's like EAGAIn one of these errors that are usually understood to be "transient", and repeating an operation could fix things. which seems like the wrong thing to communicate here.

maybe ENOTRECOVERABLE?